tests/packagedcode/data/npm/yarn-lock/v1-complex2/yarn.lock

```diff
@@ -0,0 +1,109 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+"@babel/core@^7.0.0", "@babel/core@^7.23.3", "@babel/core@^7.23.9":
+  version "7.24.4"
+  resolved "https://registry.yarnpkg.com/@babel/core/-/core-7.24.4.tgz#1f758428e88e0d8c563874741bc4ffc4f71a4717"
+  integrity sha512-MBVlMXP+kkl5394RBLSxxk/iLTeVGuXTV3cIDXavPpMMqnSnt6apKgan/U8O3USWZCWZT/TbgfEpKa4uMgN4Dg==
+  dependencies:
+    "@babel/parser" "^7.24.4"
+    "@babel/types" "^7.24.0"
+    json5 "^2.2.3"
+    semver "^6.3.1"
+
+ansi-styles@^3.2.0, ansi-styles@^3.2.1:
+  version "3.2.1"
+  resolved "https://registry.yarnpkg.com/ansi-styles/-/ansi-styles-3.2.1.tgz#41fbb20243e50b12be0f04b8dedbf07520ce841d"
+  integrity sha512-VT0ZI6kZRdTh8YyJw3SMbYm/u+NqfsAxEpWO0Pf9sq8/e94WxxOpPKx9FR1FlyCtOVDNOQ+8ntlqFxiRc+r5qA==
+  dependencies:
+    color-convert "^1.9.0"
+
+change-case@^5.4.3, change-case@^5.4.4, "latest-change-case@npm:change-case@^5.4.3":
```

Comment:
We were getting a
 ```
AssertionError: Different names for same dependency is not supported: {('', 'latest-change-case@npm:change-case'), ('', 'change-case')}
E           assert 2 == 1
E            +  where 2 = len({('', 'change-case'), ('', 'latest-change-case@npm:change-case')})
```
While trying to parse this yarn.lock file, at [this assert statement](https://github.com/nexB/scancode-toolkit/blob/fix-yarn-lock-v1-parser/src/packagedcode/npm.py#L660).

Reply:
OK, so this is resolved!